include/string.h

```diff
@@ -49,8 +49,7 @@ int memcmp (const void * : itype(_Array_ptr<const void>) byte_count(n),
 void *memchr (const void *src : itype(_Array_ptr<const void>) byte_count(n),
               int c,
               size_t n)
-  : itype(_Array_ptr<void>) byte_count(n);
-
+  : bounds(src, (_Array_ptr<char>) src + n);
```

Comment:
What's the reason for changing `itype` to fully checked?

Reply:
Actually I did not change it to fully checked. Just used the bound because the type of the return value is known implicitly but it is still type safe interface.  I will add itype for the readibility.